lib/lock_manager/worker.rb

```diff
@@ -72,15 +72,12 @@ def lock_user
 
     def polling_lock(user, reason = nil)
       sleep_duration = 1
+      return lock(user, reason) unless locked?
       loop do
-        if locked?
-          log "#{host} is locked..."
-          log "waiting #{sleep_duration} seconds."
-          sleep sleep_duration
-          sleep_duration *= 2
-        else
-          break
-        end
+        log "#{host} is locked..."
+        log "waiting #{sleep_duration} seconds."
+        sleep sleep_duration
+        sleep_duration *= 2
```

Comment:
@stahnma it looks like this never checks again to see if it's locked, so it'll just loop forever?